Gems/Multiplayer/Code/Source/NetworkEntity/EntityReplication/PropertyPublisher.cpp

```diff
@@ -75,72 +76,83 @@ namespace Multiplayer
             {
-                // This has been acked, so everything after to this and this replication record are not useful
                 mostRecentAckedIter = iter;
                 m_remoteReplicatorEstablished = true;
                 break;
             }
         }
 
-        // delete everything prior to this
+        // Delete all of the acknowledged records.
         m_sentRecords.erase(mostRecentAckedIter, m_sentRecords.end());
 
         // Nothing to send
         if (!m_pendingRecord.HasChanges() && m_sentRecords.empty() && m_remoteReplicatorEstablished)
         {
             return false;
         }
+
+        // Still need to send a change record if there are pending changes or
+        // if the remote replicator hasn't acknowledged a connection yet.
         return true;
     }
 
-    bool PropertyPublisher::PrepareAddEntityRecord()
+    void PropertyPublisher::PrepareAddEntityRecord(NetBindComponent* netBindComponent)
     {
+        AZ_Assert(netBindComponent, "NetBindComponent is nullptr");
+
+        // On an "Add", create a change record that contains all the serialized fields for the entity.
         m_sentRecords.clear();
-        m_netBindComponent->FillTotalReplicationRecord(m_pendingRecord);
+        netBindComponent->FillTotalReplicationRecord(m_pendingRecord);
         m_sentRecords.push_front(m_pendingRecord);
-        return true;
     }
 
-    bool PropertyPublisher::PrepareRebaseEntityRecord()
+    void PropertyPublisher::PrepareRebaseEntityRecord(NetBindComponent* netBindComponent)
     {
-        AZ_Assert(m_netBindComponent, "NetBindComponent is nullptr");
+        AZ_Assert(netBindComponent, "NetBindComponent is nullptr");
 
         // This is basically an Add record, but we don't want to send back predictable values
         m_sentRecords.clear();
-        m_netBindComponent->FillTotalReplicationRecord(m_pendingRecord);
+        netBindComponent->FillTotalReplicationRecord(m_pendingRecord);
         // Don't send predictable properties back to the Autonomous unless we correct them
         if (m_pendingRecord.GetRemoteNetworkRole() == NetEntityRole::Autonomous)
         {
-            m_pendingRecord.Subtract(m_netBindComponent->GetPredictableRecord());
+            m_pendingRecord.Subtract(netBindComponent->GetPredictableRecord());
         }
         m_sentRecords.push_front(m_pendingRecord);
-        return true;
     }
 
-    bool PropertyPublisher::PrepareUpdateEntityRecord()
+    void PropertyPublisher::PrepareUpdateEntityRecord(NetBindComponent* netBindComponent)
     {
-        bool didPrepare = true;
         if (m_sentRecords.size() >= net_EntityReplicatorRecordsMax)
         {
-            // If we reach the maximum outstanding records, reset the replication state
-            didPrepare = PrepareAddEntityRecord();
+            // If we reach the maximum outstanding records, reset the replication state by creating an "Add" record.
```

Comment:
This could use slightly more detail about what an Add record is. I'm guessing it's just a total reset record, all values that differ from the original prefab state?